File: game-server/src/main/java/org/forfun/mmorpg/game/player/event/PlayerLevelUpEvent.java
package org.forfun.mmorpg.game.player.event;

import jforgame.commons.eventbus.BaseEvent;
import org.forfun.mmorpg.game.database.user.entity.PlayerEnt;

public class PlayerLevelUpEvent implements BaseEvent {

	private PlayerEnt player;

	public PlayerLevelUpEvent(PlayerEnt player) {
		this.player = player;
	}

	@Override
	public PlayerEnt getOwner() {
		return player;
	}

}

File: game-server/src/main/kotlin/org/forfun/mmorpg/game/player/event/PlayerLevelUpEvent.kt
package org.forfun.mmorpg.game.player.event

import jforgame.commons.eventbus.BaseEvent
import org.forfun.mmorpg.game.database.user.entity.PlayerEnt

class PlayerLevelUpEvent(private val player: PlayerEnt) : BaseEvent {

    override fun getOwner(): PlayerEnt = player
}
